### PR TITLE
[instancer] support BASE table

### PR DESCRIPTION
**Pull request by qxliu76**
For #4161

### CODE REVIEW

File: src/hb-ot-layout-base-table.hh [hunk 1]
```diff
@@ -125,6 +125,20 @@ struct BaseCoordFormat3
     auto *out = c->serializer->embed (*this);
     if (unlikely (!out)) return_trace (false);
 
+    if (!c->plan->pinned_at_default)
+    {
+      unsigned var_idx = (this+deviceTable).get_variation_index ();
+      if (var_idx != HB_OT_LAYOUT_NO_VARIATIONS_INDEX)
```
Comment (behdad) [inline src/hb-ot-layout-base-table.hh:131]: That's not the right constant. Use `VarIdx::NO_VARIATION_INDEX` or something like that.

**Comment by behdad** [comment]
Can you rebase on main and resolve the conflict please?

**Comment by qxliu76** [comment]
> Can you rebase on main and resolve the conflict please?

Done.

**Comment by qxliu76** [comment]
That test failure seems not related with my PR

**Pull request merged**